lib/rust/lexer/definition/src/lexer.rs

```diff
@@ -464,9 +722,805 @@ impl EnsoLexer {
+        self.discard_current();
+        self.text_state.append_segment(escape);
+    }
+
+    /// Triggered when encountering a byte escape sequence.
+    fn on_escape_byte<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Text::on_escape_byte");
+        let current_match = self.consume_current();
+        let offset        = self.offset.consume();
+        let escape        = escape::ByteEscape::build(current_match,offset);
+        self.text_state.append_segment(escape);
+    }
+
+    /// Triggered when encountering a U16 unicode escape sequence.
+    fn on_escape_u16<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Text::on_escape_u16");
+        let current_match = self.consume_current();
+        let offset        = self.offset.consume();
+        let escape        = escape::U16::build(current_match,offset);
+        self.text_state.append_segment(escape);
+    }
+
+    /// Triggered when encountering a U21 unicode escape sequence.
+    fn on_escape_u21<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Text::on_escape_u32");
+        let current_match = self.consume_current();
+        let offset        = self.offset.consume();
+        let escape        = escape::U21::build(current_match,offset);
+        self.text_state.append_segment(escape);
+    }
+
+    /// Triggered when encountering a U32 unicode escape sequence.
+    fn on_escape_u32<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Text::on_escape_u32");
+        let current_match = self.consume_current();
+        let offset        = self.offset.consume();
+        let escape        = escape::U32::build(current_match,offset);
+        self.text_state.append_segment(escape);
+    }
+
+    /// Triggered when encountering an escaped raw quote.
+    fn on_escape_raw_quote<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Text::on_escape_raw_quote");
+        let offset = self.offset.consume();
+        let escape = Token::TextSegmentEscape(token::EscapeStyle::Literal,"\"",offset);
+        self.discard_current();
+        self.text_state.append_segment(escape);
+    }
+
+    /// Triggered when encountering an escaped format quote.
+    fn on_escape_format_quote<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Text::on_escape_format_quote");
+        let offset = self.offset.consume();
+        let escape = Token::TextSegmentEscape(token::EscapeStyle::Literal,"'",offset);
+        self.discard_current();
+        self.text_state.append_segment(escape);
+    }
+
+    /// Triggered when encountering an escaped interpolate quote.
+    fn on_escape_interpolate_quote<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Text::on_escape_interpolate_quote");
+        let offset = self.offset.consume();
+        let escape = Token::TextSegmentEscape(token::EscapeStyle::Literal,"`",offset);
```

Comment:
All these things here - in many, many places in this file, are repeated constants. This file should be designed in such way, that changing for example the backtick char to somethng else, it will be changed in ALL usage places. So no literal should repeat. This is very error prone currently.